scripts/production/repair_mlss_tags.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Repairs MLSS tags that may have the wrong value, may be missing, or belong to undefined MLSSs.
+
+The Method Link Species Set (MLSS) tags supported are:
+    * `max_align`
+    * `msa_mlss_id`
+
+Examples::
+
+    $ python repair_mlss_tags.py --max_align \
+        --url mysql://ensadmin:${ENSADMIN_PSW}@mysql-ens-compara-prod-6:4616/ensembl_compara_metazoa_51_104
+
+    $ python repair_mlss_tags.py --msa_mlss_id \
+        --url mysql://ensadmin:${ENSADMIN_PSW}@mysql-ens-compara-prod-5:4615/ensembl_compara_plants_51_104
+
+    $ python repair_mlss_tags.py --url $(cp1-w details url)ensembl_compara_104 --msa_mlss_id --max_align
+
+"""
+
+from argparse import ArgumentParser
+
+from ensembl.compara.db import DBConnection
+
+
+EXPECTED_VALUES_SQL = {
+    'max_align': """
+        SELECT method_link_species_set_id AS mlss_id, MAX(dnafrag_end - dnafrag_start) + 2 AS value
+        FROM constrained_element
+        GROUP BY method_link_species_set_id
+        UNION
+        SELECT method_link_species_set_id AS mlss_id, MAX(dnafrag_end - dnafrag_start) + 2 AS value
+        FROM genomic_align
+        GROUP BY method_link_species_set_id;
```

Comment:
```suggestion
        SELECT 
            method_link_species_set_id AS mlss_id,
            MAX(dnafrag_end - dnafrag_start) + 2 AS value
        FROM
            constrained_element
        GROUP BY method_link_species_set_id 
        UNION SELECT 
            method_link_species_set_id AS mlss_id,
            MAX(dnafrag_end - dnafrag_start) + 2 AS value
        FROM
            genomic_align
        GROUP BY method_link_species_set_id
```
Could be more beautiful in readability and sql style formatting :art:

Reply:
I have never liked this formatting style :sweat_smile: but let's see what the other reviewer thinks.

Reply:
Agree. My brain instantly and effortlessly processes it when formatted like this.

Reply:
The key benefit of the formatting style I have presented, I think, is that any sql changes are easily comparable in the `git` line, because it is easy to see where an additional column/table/join/where clause etc is added to, removed or swapped out :+1: - I am not too bothered either way :shrug: